### PR TITLE
Adding more TP API endpoints

### LINKED ISSUE #13: Support TP Features

### LINKED ISSUE #17: All debugLog messages should prepend [targetprocess]
This will make sure it's obvious what library is generating the log message
in other projects where there may be a lot of libraries logging out.

### PR DESCRIPTION
**Pull request by sudermanjr**
Add the following entities with `Get<Entity>`
* `CustomFields`
* `EntityState`
* `Feature` -  Fixes #13
* `Process`
* `Workflow`

Add the following set methods to UserStory:
* `SetProject` 
* `SetTeam`
* `SetFeature`

Additional Changes:
* added the ability to create and POST lists of user stories in bulk
* Added [targetprocess] to all debugLog calls. Fixes #17

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -190,6 +190,7 @@ func (c *Client) do(out interface{}, req *http.Request, urlPath string) error {
 	if err != nil {
 		return errors.Wrapf(err, "HTTP Read error on response for %s", urlPath)
 	}
+	c.debugLog(fmt.Sprintf("raw response: %s", string(b)))
```
Comment (lucasreed) [inline client.go:193]:
```suggestion
	c.debugLog(fmt.Sprintf("[targetprocess] raw response: %s", string(b)))
```

Comment (lucasreed) [reply client.go:193]: Making this suggestion here with the knowledge that we should go back and put this in to all debugLog messages. (issue #17 opened)

I was inconsistent with it in the initial creation.

File: feature.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		if err != nil {
+			return ret, err
+		}
+		ret = append(ret, innerOut.Items...)
+		out = innerOut
+	}
+	return ret, nil
+}
+
+// GetFeature will return a single feature based on its name. If somehow there are projects with the same name,
```
Comment (lucasreed) [inline feature.go:66]:
```suggestion
// GetFeature will return a single feature based on its name. If somehow there are features with the same name,
```

File: userstory.go [hunk 3]
```diff
@@ -140,3 +158,36 @@ func (us UserStory) Create() (int32, error) {
...
+		return nil, errors.Wrap(err, fmt.Sprintf("error marshaling POST body for UserStoryList %v", usl))
+	}
+	client.debugLog(fmt.Sprintf("Attempting to POST UserStory: %+v", usl))
+	err = client.Post(resp, "UserStories/bulk", nil, body)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("error POSTing UserStoryList %v", usl))
+	}
+	client.debugLog("Successfully POSTed UserStoryList")
+
+	ret := []int32{}
```
Comment (lucasreed) [inline userstory.go:187]:
```suggestion
	var ret []int32
```

Comment (lucasreed) [reply userstory.go:187]: Goland told me this is better practice 🤷

File: userstory.go [hunk 4]
```diff
@@ -140,3 +158,36 @@ func (us UserStory) Create() (int32, error) {
...
 }
+
+// NewUserStoryList returns a UserStoryList from a list of user stories.
+// Used for batch POSTing of UserStories
+func (c *Client) NewUserStoryList(list []UserStory) *UserStoryList {
+	ret := &UserStoryList{
+		client:  c,
+		Stories: list,
+	}
+	return ret
```
Comment (lucasreed) [inline userstory.go:169]:
```suggestion
	return &UserStoryList{
		client:  c,
		Stories: list,
	}
```

File: process.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package targetprocess
+
+// Process contains metadata for the state of an Process. Collection of Processs
```
Comment (lucasreed) [inline process.go:17]:
```suggestion
// Process contains metadata for the state of a Process. Collection of Processes
```

File: team.go [hunk 6]
```diff
@@ -55,24 +55,20 @@ func (c *Client) GetTeam(name string) (Team, error) {
...
-		client:      t.client,
-		Name:        name,
-		Description: description,
-	}
-	t.client.debugLog(fmt.Sprintf("Attempting to Get Project: %s", project))
-	p, err := t.client.GetProject(project)
+	us, err := NewUserStory(t.Client, name, description, project)
 	if err != nil {
-		return UserStory{}, err
+		return UserStory{}, nil
```
Comment (lucasreed) [inline team.go:70]: Should we not be returning the error here?

File: workflow.go [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package targetprocess
+
+// Workflow contains metadata for the state of an Workflow. Collection of Workflows
```
Comment (lucasreed) [inline workflow.go:17]:
```suggestion
// Workflow contains metadata for the state of an Workflow. A collection of Workflows
```

File: team.go [hunk 8]
```diff
@@ -22,7 +22,7 @@ import (
 
 // Team matches up with a targetprocess Team
 type Team struct {
-	client *Client
+	Client *Client `json:"-"`
```
[the remark this answers is not in the document]
Comment (sudermanjr) [reply team.go:25]: That was a mistake. I was trying something out. At some point we might change our minds, but I have no good arguments for that yet.
```suggestion
	client *Client
```

File: team.go (hunk 6 above)
Comment (sudermanjr) [reply team.go:70]:
```suggestion
		return UserStory{}, err
```

**Pull request merged**